include/irr/asset/IRenderpassIndependentPipeline.h

```diff
@@ -312,7 +313,7 @@ enum E_VERTEX_ATTRIBUTE_ID
 */
 
 template<typename SpecShaderType, typename LayoutType>
-class IRenderpassIndependentPipeline : public IPipeline<IRenderpassIndependentPipeline, LayoutType>
+class IRenderpassIndependentPipeline : public IPipeline<LayoutType>, public virtual core::IReferenceCounted
```

Comment:
why the virtual inheritance?

Reply:
because of parent pipeline, see comments there, it's todo tricky stuff, left because not needed for GL (buildable req)

Reply:
explain, doesn't `IPipeline` inherit from `IReferenceCounted` ?

Reply:
@Crisspl  ?

Reply:
no, it doesnt

Reply:
virtual inheritance because its base for ICPUpipeline and ICPUpipeline also inherits from IAsset and IAsset inherits from irefcounted

Reply:
shouldn't `IPipeline` inherit virtually from `IReferenceCounted` instead then?

Reply:
@Crisspl will merge when IPipeline inherits from IReferenceCounted

Reply:
fuck it I'll do it myself.

Reply:
erm sorry

Reply:
no problem, I've decided to do #148  today.